internal/plugin/grpc_provider.go

```diff
@@ -686,6 +693,96 @@ func (p *GRPCProvider) ReadDataSource(r providers.ReadDataSourceRequest) (resp p
 	return resp
 }
 
+func (p *GRPCProvider) CallFunction(r providers.CallFunctionRequest) (resp providers.CallFunctionResponse) {
+	logger.Trace("GRPCProvider: CallFunction")
+
+	schema := p.GetProviderSchema()
+	if schema.Diagnostics.HasErrors() {
+		// This should be unreachable
+		resp.Error = schema.Diagnostics.Err()
+		return resp
+	}
+
+	spec, ok := schema.Functions[r.Name]
+	if !ok {
+		// This should be unreachable
+		resp.Error = fmt.Errorf("invalid CallFunctionRequest: function %s not defined in provider schema", r.Name)
+		return resp
+	}
+
+	protoReq := &proto.CallFunction_Request{
+		Name:      r.Name,
+		Arguments: make([]*proto.DynamicValue, len(r.Arguments)),
+	}
+
+	// Translate the arguments
+	// As this is functionallity is always sitting behind cty/function.Function, we skip some validation
```

Comment:
I think there's a stray "is" in that sentence and a stray "l" in functionality.

Reply:
Fixed